corehq/apps/export/static/export/js/list_exports.ng.js

```diff
@@ -116,6 +127,99 @@
                     }
                 });
         };
+        $scope.setFilterModalExport = function (export_) {
+            // The filterModalExport is used as context for the FeedFilterFormController
+            $rootScope.filterModalExport = export_;
+        };
+        $scope.isLocationSafeForUser = function(export_) {
+            return (!export_.emailedExport) || export_.emailedExport.isLocationSafeForUser;
+        };
+    };
+    exportsControllers.FeedFilterFormController = function (
+        $scope, $rootScope, djangoRMI, filterFormElements, filterFormModalElement
+    ) {
+        var self = {};
+        $scope._ = _;   // make underscore.js available
+        $scope.formData = {};
+        $scope.modelType = null;  // "form" or "case" - corresponding to the type of export selected.
+        // A list a location names. The export will be restricted to these locations.
+        $scope.locationRestrictions = [];
+        $scope.isSubmittingForm = false;
+        $scope.hasFormSubmitError = false;
+        $scope.formSubmitErrorMessage = null;
+        $scope.dateRegex = '[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]';
```

Comment:
I can't find where this is used, but you can use momentjs to do more robust date checking: https://momentjs.com/docs/#/parsing/string-format/